### PR TITLE
Expose GetBlockIndexGFXR() from capture layer

### PR DESCRIPTION
**Pull request by panos-lunarg**
GetBlockIndexGFXR() returns the current block index being processed by the capture manager.

### CODE REVIEW

**Comment by locke-lunarg** [comment]
The code looks good to me. Could you please give me some examples about where we will need the capture block index? Thank you. Plus, this capture block index is different with the decoder block index since the capture's incudes system code. The decoder's has only Vulkan code. Should we care about it? or it doesn't matter?

**Comment by locke-lunarg** [comment]
System code I mean something aren't Vulkan function calls. It doesn't matter. LGTM.

**Review by locke-lunarg — APPROVED** [review]

**Comment by locke-lunarg** [comment]
I'm thinking that could we write this block index to the file? So the decoder could just read the block index to ensure both have the same block index.

**Comment by panos-lunarg** [comment]
I am not sure if it's worth it. We can't store the id in the block header without breaking backward compatibility, so we'll have to store it as a metadata? This would essentially double the blocks written to the file.
Can you think of a better way?

**Comment by locke-lunarg** [comment]
You are right. It's not worth. We shouldn't break backward compatibility.

**Review by locke-lunarg — APPROVED** [review]

File: framework/encode/capture_manager.cpp [hunk 1]
```diff
@@ -823,21 +828,26 @@ void CaptureManager::WriteDisplayMessageCmd(const char* message)
...
         message_cmd.meta_header.block_header.type = format::BlockType::kMetaDataBlock;
         message_cmd.meta_header.block_header.size = format::GetMetaDataBlockBaseSize(message_cmd) + message_length;
         message_cmd.meta_header.meta_data_id =
             format::MakeMetaDataId(api_family_, format::MetaDataType::kDisplayMessageCommand);
-        message_cmd.thread_id = GetThreadData()->thread_id_;
+        message_cmd.thread_id = thread_data->thread_id_;
 
         CombineAndWriteToFile({ { &message_cmd, sizeof(message_cmd) }, { message, message_length } });
+
+        ++block_index_;
```
Comment (bradgrantham-lunarg) [inline framework/encode/capture_manager.cpp:843]: Is there something causing these two lines to be atomic?

**Review by bradgrantham-lunarg — APPROVED** [review]
This looks pretty good to me.  Later we can consider whether this should be exposed through some kind of "LUNARG" Vulkan extension, maybe even make it generic like VK_LUNARG_get_tooling_marker or some such.

**Pull request merged**